lib/http/HttpServer.ts

```diff
@@ -67,15 +67,22 @@ class HttpServer {
       res.writeHead(200, { 'Content-Type': 'application/json' });
       res.end(JSON.stringify(resJson));
     } catch (err) {
-      if (err.code === errorCodes.INVALID_ARGUMENT) {
-        res.writeHead(400);
-        res.end(err.message);
-      } else if (err.code === swapErrorCodes.PAYMENT_HASH_NOT_FOUND) {
-        res.writeHead(404);
-        res.end(err.message);
-      } else {
-        res.writeHead(500);
-        res.end(JSON.stringify(err));
+      switch (err.code) {
+        case errorCodes.INVALID_ARGUMENT:
+          res.writeHead(400);
+          res.end(err.message);
+          break;
+        case swapErrorCodes.INVALID_RESOLVE_REQUEST:
+          res.writeHead(400);
+          res.end(err.message);
+          break;
+        case swapErrorCodes.PAYMENT_HASH_NOT_FOUND:
+          res.writeHead(404);
+          res.end(err.message);
+          break;
+        default:
+          res.writeHead(500);
+          res.end(JSON.stringify(err));
```

Comment:
So a `break` here for the last case is not technically necessary, but I kind of like it anyway in case the cases ever get reordered or if we add another case at the end and forget to add the `break`.

There's actually a [tslint rule](https://palantir.github.io/tslint/rules/switch-final-break/) for this I just found that I think might be worth adding, I believe we have the final `break` in use elsewhere in the code so it would only enforce this rule going forward.